### PR TITLE
docs(root): add smc tss examples

### PR DESCRIPTION
**Pull request by alebusse**
added smc tss examples

WP-313

### CODE REVIEW

**Review by OttoAllmendinger — COMMENTED** [review]
flush

**Review by zahin-mohammad — DISMISSED** [review]
can u also work with @vmccarty to add these to the dev portal?

File: examples/ts/tss-smc/HOWTO.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# HOW TO
+
+## Create an EdDSA TSS Self-Managed Cold Wallet
+
+This guide explains the steps required to create an EdDSA TSS self-managed cold wallet using the provided repository.
```
Comment (vmccarty) [inline examples/ts/tss-smc/HOWTO.md:5]:
```suggestion
This guide explains how to create a self-managed TSS cold wallet for EdDSA assets using the provided repository.
```

File: examples/ts/tss-smc/HOWTO.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+## Prerequisites
+
+Before starting the process, ensure the following prerequisites are met:
+
```
Comment (vmccarty) [inline examples/ts/tss-smc/HOWTO.md:10]:
```suggestion
```

File: examples/ts/tss-smc/HOWTO.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+- Have two OVCs (Online Voting Centers) instances running, one for the User and one for the Backup.
```
Comment (vmccarty) [inline examples/ts/tss-smc/HOWTO.md:11]:
```suggestion
- Set up 2 air-gapped machines with the BitGo Offline Vault Console (OVC) (1 for the user key and 1 for the backup key).
```

File: examples/ts/tss-smc/HOWTO.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+- Node.js and npm (Node Package Manager) should be installed locally.
```
Comment (vmccarty) [inline examples/ts/tss-smc/HOWTO.md:12]:
```suggestion
- Install Node.js
- Install Node Package Manager (npm)
```

File: examples/ts/tss-smc/HOWTO.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+- Run `npm install` from the root directory of the repository to install the necessary dependencies.
```
Comment (vmccarty) [inline examples/ts/tss-smc/HOWTO.md:13]:
```suggestion
- Install the necessary dependencies by running `npm install` from the root directory of the repository.
```

File: examples/ts/tss-smc/HOWTO.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+## Steps
+
+Follow these steps:
+
```
Comment (vmccarty) [inline examples/ts/tss-smc/HOWTO.md:18]:
```suggestion
```

File: examples/ts/tss-smc/HOWTO.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+1. Open OVC 1 and select `Generate TSS Key Share` from the menu. Choose OVC 1 as the instance.
```
Comment (vmccarty) [inline examples/ts/tss-smc/HOWTO.md:19]:
```suggestion
1. Open OVC on one of the air-gapped machines. Select `Generate TSS Key Share` from the menu. Select `OVC 1` as the instance.
```

File: examples/ts/tss-smc/HOWTO.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+2. Open OVC 2 and select `Generate TSS Key Share` from the menu. Choose OVC 2 as the instance.
```
Comment (vmccarty) [inline examples/ts/tss-smc/HOWTO.md:20]:
```suggestion
2. Open the OVC on the other the air-gapped machine. Select `Generate TSS Key Share` from the menu. Select `OVC 2` as the instance.
```

File: examples/ts/tss-smc/HOWTO.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+**Do not close the OVCs after starting the process**
```
Comment (vmccarty) [inline examples/ts/tss-smc/HOWTO.md:21]:
```suggestion
**Note:** Keep both instances of the OVC open until you finish creating the wallet.
```

File: examples/ts/tss-smc/HOWTO.md [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+3. In OVC 1, follow the instructions and choose the "Single Private Key" option. Download the file required for OVC 2.
```
Comment (vmccarty) [inline examples/ts/tss-smc/HOWTO.md:22]:
```suggestion
3. In OVC 1, select the `Single Private Key` option and download the JSON.
```

File: examples/ts/tss-smc/HOWTO.md [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+4. In OVC 2, upload the file received from OVC 1 and choose the "Single Private Key" option. Download the json file required for the step 5.
```
Comment (vmccarty) [inline examples/ts/tss-smc/HOWTO.md:23]:
```suggestion
4. In OVC 2, upload the JSON and select the `Single Private Key` option. Download the updated JSON file.
```

File: examples/ts/tss-smc/HOWTO.md [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+5. Open the script `create-smc-wallet-step-1` and follow the instructions. Execute it using `ts-node create-smc-wallet-step-1.ts`. This will generate a file named `toOVC1-step-1-timestamp.json` (required for step 6). Additionally, note down the Bitgo Key Id displayed in the console.
```
Comment (vmccarty) [inline examples/ts/tss-smc/HOWTO.md:24]:
```suggestion
5. Open the script `create-smc-wallet-step-1` and follow the instructions. Execute it using `ts-node create-smc-wallet-step-1.ts`. This generates `toOVC1-step-1-timestamp.json`. Make note of the BitGo key ID displayed in the console.
```

File: examples/ts/tss-smc/HOWTO.md [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+6. Go back to OVC 1, upload the JSON file generated in the previous step (5), and proceed with the process. Download the JSON file required for OVC 2 (needed for step 7). Also, download the private key file (**store it securely!**) and the public key file (required for step 8).
```
Comment (vmccarty) [inline examples/ts/tss-smc/HOWTO.md:25]:
```suggestion
6. In OVC 1, upload the JSON file generated in the previous step. Download the updated JSON, the private-key file, and the public-key file.
**Note:** Store the private-key file in a secure location.
```

File: examples/ts/tss-smc/HOWTO.md [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+7. Go back to OVC 2, upload the JSON file generated in step 6, and continue the process. Download the private key file (**store it securely!**) and the public key file (required for step 8).
```
Comment (vmccarty) [inline examples/ts/tss-smc/HOWTO.md:26]:
```suggestion
7. In OVC 2, upload the JSON generated previous step. Download the private-key file and the public-key file.
**Note:** Store the private-key file in a secure location.
```

File: examples/ts/tss-smc/HOWTO.md [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+8. Open the script `create-smc-wallet-step-2` and follow the instructions. Ensure that you include the public key files from OVC 1 and OVC 2. Additionally, provide the Bitgo Key Id obtained in step 5. Execute the script using `ts-node create-smc-wallet-step-2.ts`. This will create a wallet, and the relevant information will be stored in a file named `wallet-walletName.json` (for informational purposes only).
```
Comment (vmccarty) [inline examples/ts/tss-smc/HOWTO.md:27]:
```suggestion
8. Open the script `create-smc-wallet-step-2` and follow the instructions. Include the public-key files from OVC 1 and OVC 2 and the BitGo key ID from step 5. Execute the script using `ts-node create-smc-wallet-step-2.ts`. This creates the wallet, and stores the relevant information in `wallet-walletName.json` (for informational purposes only).
```

File: examples/ts/tss-smc/README.md [hunk 16]
```diff
@@ -0,0 +1,23 @@
+# TSS Self-Managed Cold Wallet Examples
```
Comment (vmccarty) [inline examples/ts/tss-smc/README.md:1]:
```suggestion
# Self-Managed TSS Cold Wallet Examples
```

File: examples/ts/tss-smc/README.md [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+
+## Prerequisites
+
+ You can find information on how to do this in the root README.md.
```
Comment (vmccarty) [inline examples/ts/tss-smc/README.md:5]:
```suggestion
See README.md in the root.
```

File: examples/ts/tss-smc/create-smc-wallet-step-1.ts [hunk 18]
```diff
@@ -0,0 +1,34 @@
+import { BitGo } from "bitgo";
+import { readFileSync, writeFileSync } from "fs";
+
+// This step will create the Bitgo Key from the OVC shares
```
Comment (vmccarty) [inline examples/ts/tss-smc/create-smc-wallet-step-1.ts:4]:
```suggestion
// This creates the BitGo Key from the OVC shares
```

File: examples/ts/tss-smc/create-smc-wallet-step-1.ts [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+// Make sure to put the file from the OVC 2 in the json folder and name it fromOVC2-step-1.json
```
Comment (vmccarty) [inline examples/ts/tss-smc/create-smc-wallet-step-1.ts:5]:
```suggestion
// Put the file from OVC 2 in the JSON folder and name it fromOVC2-step-1.json
```

File: examples/ts/tss-smc/create-smc-wallet-step-1.ts [hunk 20]
```diff
@@ -0,0 +1,34 @@
...
+// It will output a file to be used in the OVC 1
```
Comment (vmccarty) [inline examples/ts/tss-smc/create-smc-wallet-step-1.ts:6]:
```suggestion
// This outputs a JSON that you use in OVC 1
```

File: examples/ts/tss-smc/create-smc-wallet-step-2.ts [hunk 21]
```diff
@@ -0,0 +1,77 @@
+import { BitGo, KeyType } from "bitgo";
+import { readFileSync, writeFileSync } from "fs";
+
+// This step will create the SMC wallet after the OVCs created both the User and Backup keys
```
Comment (vmccarty) [inline examples/ts/tss-smc/create-smc-wallet-step-2.ts:4]:
```suggestion
// This creates a self-managed cold wallet
```

File: examples/ts/tss-smc/create-smc-wallet-step-2.ts [hunk 22]
```diff
@@ -0,0 +1,77 @@
...
+// It will validate that the commonKeychain match between the OVCs and BitGo
```
Comment (vmccarty) [inline examples/ts/tss-smc/create-smc-wallet-step-2.ts:5]:
```suggestion
// This validates that the commonKeychain matches between OVC 1, OVC 2, and BitGo
```

File: examples/ts/tss-smc/create-smc-wallet-step-2.ts [hunk 23]
```diff
@@ -0,0 +1,77 @@
...
+// Make sure to set the bitgo key id from the step 1
```
Comment (vmccarty) [inline examples/ts/tss-smc/create-smc-wallet-step-2.ts:6]:
```suggestion
// Set the BitGo key ID from the step 1
```

File: examples/ts/tss-smc/create-smc-wallet-step-2.ts [hunk 24]
```diff
@@ -0,0 +1,77 @@
...
+// Make sure to put the public key file from the OVC 1 in the json folder and name it public-key-from-ovc1.json
```
Comment (vmccarty) [inline examples/ts/tss-smc/create-smc-wallet-step-2.ts:7]:
```suggestion
// Put the public key file from the OVC 1 in the JSON folder and name it public-key-from-ovc1.json
```

File: examples/ts/tss-smc/create-smc-wallet-step-2.ts [hunk 25]
```diff
@@ -0,0 +1,77 @@
...
+// Make sure to put the public key file from the OVC 2 in the json folder and name it public-key-from-ovc2.json
```
Comment (vmccarty) [inline examples/ts/tss-smc/create-smc-wallet-step-2.ts:8]:
```suggestion
// Put the public key file from the OVC 2 in the JSON folder and name it public-key-from-ovc2.json
```

File: examples/ts/tss-smc/create-smc-wallet-step-2.ts [hunk 26]
```diff
@@ -0,0 +1,77 @@
...
+// It will output a file with the wallet information
```
Comment (vmccarty) [inline examples/ts/tss-smc/create-smc-wallet-step-2.ts:9]:
```suggestion
// This outputs a file with the wallet information
```

**Review by vmccarty — COMMENTED** [review]
Left minimalism suggestions for the instructional test. Well written. Nice job!

**Review by zahin-mohammad — APPROVED** [review]

**Pull request merged**